stl/inc/__msvc_bit_utils.hpp

```diff
@@ -140,9 +140,26 @@ _NODISCARD int _Checked_x86_x64_countl_zero(const _Ty _Val) noexcept {
 
 #if defined(_M_ARM) || defined(_M_ARM64)
 #ifdef __clang__ // TRANSITION, GH-1586
+#ifdef _M_ARM64
+_NODISCARD constexpr int _Clang_arm_arm64_countl_zero(const unsigned char _Val) {
+    // TODO: Revert back to __builtin_clzs once clang longer generates worse code than __builtin_clz
+    return __builtin_clz(_Val) - (_Unsigned_integer_digits<unsigned int> - _Unsigned_integer_digits<unsigned char>);
+}
+
+_NODISCARD constexpr int _Clang_arm_arm64_countl_zero(const unsigned short _Val) {
+    // TODO: Revert back to __builtin_clzs once clang longer generates worse code than __builtin_clz
+    return __builtin_clz(_Val) - (_Unsigned_integer_digits<unsigned int> - _Unsigned_integer_digits<unsigned short>);
+}
+#else
+_NODISCARD constexpr int _Clang_arm_arm64_countl_zero(const unsigned char _Val) {
+    // TODO: Revert back to __builtin_clzs once clang longer generates worse code than __builtin_clz
+    return __builtin_clzs(_Val) - (_Unsigned_integer_digits<unsigned short> - _Unsigned_integer_digits<unsigned char>);
+}
+
 _NODISCARD constexpr int _Clang_arm_arm64_countl_zero(const unsigned short _Val) {
     return __builtin_clzs(_Val);
 }
+#endif

```

Comment:
This `#endif` should be commented. (We add preprocessor comments for everything beyond 1-2 lines to help keep track of what region we're in.)